src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -257,133 +263,199 @@ private void enqueueNotification(TripMonitorNotification ...tripMonitorNotificat
             // It's been about an hour since the last check. Do not skip.
             if (minutesSinceLastCheck >= 60) {
                 // TODO: Change log level.
-                LOG.info("Trip {} not checked in at least an hour. Checking.", trip.id);
+                LOG.info("Trip not checked in at least an hour. Checking.");
                 return false;
             }
         } else {
             // It's less than an hour until the trip time, start more frequent trip checks (about every 15 minutes).
             if (minutesSinceLastCheck >= 15) {
                 // Last check was more than 15 minutes ago. Check. (approx. 4 checks per hour).
                 // TODO: Change log level.
-                LOG.info("Trip {} happening soon. Checking.", trip.id);
+                LOG.info("Trip happening soon. Checking.");
                 return false;
             }
-            // If the minutes until the trip is within the lead time, check the trip every minute
-            // (assuming the loop runs every minute).
-            // TODO: Should lead time just be used for notification time limit?
-            if (minutesUntilTrip <= trip.leadTimeInMinutes) {
+            // If the trip starts within 30 minutes, check the trip every minute (assuming the loop runs every minute).
+            if (minutesUntilTrip <= 30) {
                 // TODO: Change log level.
-                LOG.info(
-                    "Trip {} happening within user's lead time ({} minutes). Checking every minute.",
-                    trip.id,
-                    trip.leadTimeInMinutes
-                );
+                LOG.info("Trip happening within 30 minutes. Checking every minute.");
                 return false;
             }
         }
         // TODO: Check that journey state is not flagged
         // TODO: Check last notification time.
         // Default to skipping check.
         // TODO: Change log level.
-        LOG.info("Trip {} criteria not met to check. Skipping.", trip.id);
+        LOG.info("Trip criteria not met to check. Skipping.");
         return true;
     }
 
     /**
-     * Generate the appropriate OTP query params for the trip for the current check. This currently involves replacing
+     * Calculate the next possible time that the itinerary is possible, advancing the targetZonedDateTime and setting
+     * the targetZonedDateTime, targetDate, otpResponse and matchingItinerary. Returns false if an error occurs or a
+     * matching itinerary isn't found on the next possible date the trip should be monitored on.
+     */
+    private boolean calculateNextItinerary() {
+        // if the trip is not active on the current zoned date time, advance until a day is found when the trip is
+        // active. It is guaranteed that the trip will be active on a certain date because a call to
+        // {@link MonitoredTrip#isInactive} should already have been made in the shouldSkipMonitoredTripCheck method.
```

Comment:
This guarantee is valid only if this method is only called in the correct location (otherwise there is no guarantee and this could result in an endless loop). Instead of making comments like this, I think a better approach is just to make a call to isInactive in this method body just to make sure.

Reply:
Added double-checking code in 6fd00a976c063c5b46e7254a995fbbb628b90445